### PR TITLE
Add Appknox starter workflow

### PR DESCRIPTION
**Pull request by ginilpg**
## Pre-requisites

- [x] Prior to submitting a new workflow, please apply to join the GitHub Technology Partner Program: [partner.github.com/apply](https://partner.github.com/apply?partnershipType=Technology+Partner).

---

### **Please note that at this time we are only accepting new starter workflows for Code Scanning. Updates to existing starter workflows are fine.**

---

## Tasks

**For _all_ workflows, the workflow:**

- [x] Should be contained in a `.yml` file with the language or platform as its filename, in lower, [_kebab-cased_](https://en.wikipedia.org/wiki/Kebab_case) format (for example, [`docker-image.yml`](https://github.com/actions/starter-workflows/blob/main/ci/docker-image.yml)).  Special characters should be removed or replaced with words as appropriate (for example, "dotnet" instead of ".NET").
- [x ] Should use sentence case for the names of workflows and steps (for example, "Run tests").
- [x] Should be named _only_ by the name of the language or platform (for example, "Go", not "Go CI" or "Go Build").
- [x] Should include comments in the workflow for any parts that are not obvious or could use clarification.
- [x] Should specify least privileged [permissions](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#modifying-the-permissions-for-the-github_token) for `GITHUB_TOKEN` so that the workflow runs successfully.

**For _CI_ workflows, the workflow:**

- [ ] Should be preserved under [the `ci` directory](https://github.com/actions/starter-workflows/tree/main/ci).
- [ ] Should include a matching `ci/properties/*.properties.json` file (for example, [`ci/properties/docker-publish.properties.json`](https://github.com/actions/starter-workflows/blob/main/ci/properties/docker-publish.properties.json)).
- [ ] Should run on `push` to `branches: [ $default-branch ]` and `pull_request` to `branches: [ $default-branch ]`.
- [ ] Packaging workflows should run on `release` with `types: [ created ]`.
- [ ] Publishing workflows should have a filename that is the name of the language or platform, in lower case, followed by "-publish" (for example, [`docker-publish.yml`](https://github.com/actions/starter-workflows/blob/main/ci/docker-publish.yml)).

**For _Code Scanning_ workflows, the workflow:**

- [x] Should be preserved under [the `code-scanning` directory](https://github.com/actions/starter-workflows/tree/main/code-scanning).
- [x] Should include a matching `code-scanning/properties/*.properties.json` file (for example, [`code-scanning/properties/codeql.properties.json`](https://github.com/actions/starter-workflows/blob/main/code-scanning/properties/codeql.properties.json)), with properties set as follows:
  - [x] `name`: Name of the Code Scanning integration.
  - [x] `creator`: Name of the organization/user producing the Code Scanning integration.
  - [x] `description`: Short description of the Code Scanning integration.
  - [x] `categories`: Array of languages supported by the Code Scanning integration.
  - [x] `iconName`: Name of the SVG logo representing the Code Scanning integration. This SVG logo must be present in [the `icons` directory](https://github.com/actions/starter-workflows/tree/main/icons).
- [x] Should run on `push` to `branches: [ $default-branch, $protected-branches ]` and `pull_request` to `branches: [ $default-branch ]`. We also recommend a `schedule` trigger of `cron: $cron-weekly` (for example, [`codeql.yml`](https://github.com/actions/starter-workflows/blob/c59b62dee0eae1f9f368b7011cf05c2fc42cf084/code-scanning/codeql.yml#L14-L21)).

**Some general notes:**

- [x] This workflow must _only_ use actions that are produced by GitHub, [in the `actions` organization](https://github.com/actions), **or**
- [x] This workflow must _only_ use actions that are produced by the language or ecosystem that the workflow supports.  These actions must be [published to the GitHub Marketplace](https://github.com/marketplace?type=actions).  We require that these actions be referenced using the full 40 character hash of the action's commit instead of a tag.  Additionally, workflows must include the following comment at the top of the workflow file:
    ```
    # This workflow uses actions that are not certified by GitHub.
    # They are provided by a third-party and are governed by
    # separate terms of service, privacy policy, and support
    # documentation.
    ```
- [x] Automation and CI workflows should not send data to any 3rd party service except for the purposes of installing dependencies.
- [x] Automation and CI workflows cannot be dependent on a paid service or product.

### CODE REVIEW

**Review by ALI12125555 — APPROVED** [review]

**Review by subho007 — APPROVED** [review]

**Comment by ginilpg** [comment]
@nickfyson Could you please assist us with the merging process? It has been pending for almost a month. Thank you

File: code-scanning/appknox.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        with:
+          java-version: 1.8
+      - name: Grant execute permission for gradlew
+        run: chmod +x gradlew
+        
+      - name: Build the app
+        run: ./gradlew build                                         # Update this to build your Android or iOS application
+        
+      - name: Appknox GitHub action
+        uses: appknox/appknox-github-action@1.1.0
```
Comment (alexisabril) [inline code-scanning/appknox.yml:44]: Would you be able to provide the hash instead of the version for this Action?

**Review by alexisabril — CHANGES_REQUESTED** [review]
Thank you @ginilpg for the PR! The only request I noted was for a hash to the requested Action.

**Review by Limbersky23 — APPROVED** [review]

**Review by Limbersky23 — APPROVED** [review]

**Review by Limbersky23 — APPROVED** [review]

File: code-scanning/appknox.yml (hunk 1 above)
Comment (ginilpg) [reply code-scanning/appknox.yml]: @alexisabril I have added the hash here. Please verify.

**Comment by ginilpg** [comment]
@alexisabril We have addressed the review comment you provided. Could you please reverify the changes?

**Comment by ginilpg** [comment]
@alexisabril Could you please assist us with the merging process here?

**Comment by subho007** [comment]
@alexisabril let us know if there is any more changes required ?

**Comment by alexisabril** [comment]
Hi folks, taking a look at this PR it looks as if there might be a formatting issue in the yml for this action.

**Review by ginilpg — COMMENTED** [review]
Yml formatting issues fixed. @alexisabril Can you please verify the changes?

**Comment by ginilpg** [comment]
> Hi folks, taking a look at this PR it looks as if there might be a formatting issue in the yml for this action.

@alexisabril Formatting issues has been fixed. Could you please verify the changes?

**Review by alexisabril — APPROVED** [review]

**Comment by ginilpg** [comment]
@alexisabril Could you please help us on the next steps here? It looks like the workflow needs your approval to run

**Comment by subho007** [comment]
Hi @alexisabril, thank you for taking the time to review the PR. Could you please let us know when we can proceed with going live?

**Comment by alexisabril** [comment]
Hi folks, it appears this PR just needs to be updated at this point.

**Comment by subho007** [comment]
@ginilpg can you rebase and update the branch?

**Comment by ginilpg** [comment]
> Hi folks, it appears this PR just needs to be updated at this point.

@alexisabril The PR has been updated

**Comment by ginilpg** [comment]
@alexisabril  Thanks for merging the PR. We noticed that the 'Code Scanning' category is missing under workflows now. Can we add the 'Security' also as a category along with code scanning? Additionally, I realised we missed removing the 'preview' label from the properties.json file, so I've created a new PR to address that as well. Apologies for the back and forth iterations

**Review by alphachart — COMMENTED** [review]
Check

**Pull request merged**